glambie/data/timeseries_data.py

```diff
@@ -26,21 +30,42 @@ def max_change_value(self) -> float:
         finite_list = np.isfinite(self.changes)
         return np.max(self.changes[finite_list])
 
-    def __init__(self, dates: np.ndarray, area: np.ndarray,
-                 changes: np.ndarray, errors: np.ndarray, rgi_version: int, region_id: int, unit: str,
-                 user: Optional[str] = None, user_group: Optional[str] = None, data_group: Optional[str] = None):
+    def __init__(self, data_group: Optional[GlambieDataGroup] = None, region: Optional[RGIRegion] = None,
+                 rgi_version: Optional[int] = None, unit: Optional[str] = None,
+                 data_filepath: Optional[str] = None, user: Optional[str] = None, user_group: Optional[str] = None,
+                 read_data: Optional[bool] = False):
         # name of user
         self.user = user
         # experiment group
         self.user_group = user_group
         # experiment group (from data files)
         self.data_group = data_group
-        # zwally/rignot/generic
+        # rgi
         self.rgi_version = rgi_version
-        # basin/ice-sheet id
-        self.basin_id = region_id
+        # basin
+        self.region = region
         # unit of timeseries, ie mwe (gravimetry) or m (other)
         self.unit = unit
+        # filepath of data
+        self.data_filepath = data_filepath
+        if read_data and self.data_filepath is not None:
+            self.read_data()
+
+    def read_data(self, data_filepath: Optional[str] = None):
```

Comment:
Let's have a chat about this pattern in general. I feel like there's a cleaner way to capture what you're trying to achieve. For my own benefit later: The class has some basic info and then some more information that's populated with ingest_data. But you want the contents to lazy load. It's probably better to have a class state flag self.is_ingesed. And then whenever you need the data have a method ingest_if_necessary() that checks if it's already ingested or not and loads it. Maybe - let's discuss :-)